File: android-project/app/src/main/java/com/example/mada_2/server_connection/ML.java
package com.example.mada_2.server_connection;

import java.io.*;
import okhttp3.*;

public class ML {
    public static String getPrediction(android.net.Uri imageUri) throws IOException{
        OkHttpClient client = new OkHttpClient().newBuilder()
                .build();
        MediaType mediaType = MediaType.parse("text/plain");
        RequestBody body = new MultipartBody.Builder().setType(MultipartBody.FORM)
                .addFormDataPart("file","123",
                        RequestBody.create(MediaType.parse("application/octet-stream"),
                                new File(imageUri.getPath())))
                .build();
        Request request = new Request.Builder()
                .url("127.0.0.1:8000/recognize_meters_data")
                .method("POST", body)
                .build();
        Response response = client.newCall(request).execute();
        return response.body().string();
    }
}

File: android-project/app/src/main/java/com/example/mada_2/ML.kt
package com.example.mada_2

import android.net.Uri
import okhttp3.*
import java.io.File
import java.io.IOException

class ML {
    @Throws(IOException::class)
    fun getPrediction(imageUri: Uri): String? {
        val client = OkHttpClient().newBuilder()
            .build()
        val mediaType: MediaType = parse.parse("text/plain")
        val body: RequestBody = Builder().setType(MultipartBody.FORM)
            .addFormDataPart(
                "file", "123",
                RequestBody.create(
                    parse.parse("application/octet-stream"),
                    File(imageUri.path)
                )
            )
            .build()
        val request: Request = Builder()
            .url("127.0.0.1:8000/recognize_meters_data")
            .method("POST", body)
            .build()
        val response = client.newCall(request).execute()
        return response.body!!.string()
    }
}